packages/app/src/server/service/search-delegator/elasticsearch.ts

```diff
@@ -144,7 +147,10 @@ class ElasticsearchDelegator implements SearchDelegator<Data> {
   }
 
   async init() {
-    return this.normalizeIndices();
+    if (!this.isElasticsearchReindexOnBoot) {
+      return this.normalizeIndices(); // call normalizeIndices() only
+    }
+    return this.rebuildIndex(); // rebuildIndex() will call normalizeIndeces() in the end of rebuild process
```

Comment:
normalizeIndices は、isElasticsearchReindexOnBoot の値に依らず実行し、
その後で isElasticsearchReindexOnBoot が true なら rebuildIndex を行ってください。await を利用して構いません。

## 想定
この init 自体、利用する側では await せずに呼んでいるので、結局非同期実行になるはずです。
rebuildIndex に時間がかかるとしても、その完了を待たず SearchService 自体の初期化は完了する想定です。
この想定と違う動きをする場合は相談をください。

Reply:
Execute `normalizeIndices` regardless of the value of `isElasticsearchReindexOnBoot`.
After that, if `isElasticsearchReindexOnBoot` is true, execute rebuildIndex.
You can use await there.

## Expectation

This `init` should become asynchronous because it is used without await.
Although rebuildIndex may take long time, I still expect that initialization of SearchService will be complete before rebuildIndex is finished.
Please consult me if the actual behavior is different from this expectation.